storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -72,8 +72,12 @@ class Store implements UpdatableStore {
 
   private final int hotStatePersistenceFrequencyInEpochs;
 
+  private final ReadWriteLock votesLock = new ReentrantReadWriteLock();
+  private final Lock readVotesLock = votesLock.readLock();
   private final ReadWriteLock lock = new ReentrantReadWriteLock();
   private final Lock readLock = lock.readLock();
+  private final ReadWriteLock timingLock = new ReentrantReadWriteLock();
+  private final Lock readTimingLock = timingLock.readLock();
```

Comment:
I'm somewhat nervous about the timing lock being separate because the current time is a really important concept and things like onTick update both the time and current justified checkpoint but they're now under separate locks and so may become visible at different times. It may be ok but much less clear than the separate votes lock.